### PR TITLE
Deprecate ANA reader/writer functions

### PR DESCRIPTION
**Pull request by nabobalis**
C code is old, the wrapper uses numpy deprecated API, it isn't cross platform.

It supports a file format that isn't used widely.

### CODE REVIEW

File: sunpy/io/ana.py [hunk 1]
```diff
@@ -24,11 +25,10 @@
     "C extension for ANA is missing. For more details see: "
     "https://docs.sunpy.org/en/stable/installation.html#installing-without-conda"
 )
-
-
 __all__ = ['read', 'get_header', 'write']
 
 
+@deprecated("6.0", message="The ANA reader is un-maintained and will be removed in the future.")
```
[the remark this answers is not in the document]
Comment (wtbarnes) [reply sunpy/io/ana.py:31]: I'm not terribly opposed to this but it does go against our deprecation policy of always providing an alternative when we deprecate/remove.

[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py:31]: Is that our policy?!

**Review by wtbarnes — APPROVED** [review]
Let's clean up the PR description first though.

**Comment by ayshih** [comment]
I'm not a fan of deprecating the ANA code without first attempting to find out from the community how many people - if any - are using it.  Alternatively, I ask that the deprecation message includes "contact us if you are using this".

**Comment by nabobalis** [comment]
> I'm not a fan of deprecating the ANA code without first attempting to find out from the community how many people - if any - are using it. Alternatively, I ask that the deprecation message includes "contact us if you are using this".

I can add that.

**Comment by nabobalis** [comment]
I have updated it.

File: sunpy/io/ana.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/io/ana.py]: I recalled there is one by the OSLO group, so I added a link to it.

**Comment by Cadair** [comment]
I am still a hard :-1: on this.

**Comment by Cadair** [comment]
Also this PR did not have and should not have had the minor fix label, so should have had two reviews. Please follow our dev guidelines.

**Pull request merged**